packages/lesswrong/server/scripts/backfillParentTags.ts

```diff
@@ -0,0 +1,49 @@
+import TagRels from "../../lib/collections/tagRels/collection";
+import Tags from "../../lib/collections/tags/collection";
+import { Globals, updateMutator } from "../vulcan-lib";
+import { v4 as uuid } from 'uuid';
+import { updatePostDenormalizedTags } from "../tagging/tagCallbacks";
+
+const backfillParentTags = async (childTagSlug, parentTagSlug) => {
+  const parentTag = (await Tags.find({slug: parentTagSlug}).fetch())[0];
+  const childTag = (await Tags.find({slug: childTagSlug}).fetch())[0];
+  const parentTagRelPostIds = (await TagRels.find({tagId: parentTag._id}).fetch()).map(rel => rel.postId);
+
+  const childTagRelPostIds = (await TagRels.find({tagId: childTag._id}).fetch()).filter(rel => !parentTagRelPostIds.includes(rel.postId)).map(rel => rel.postId);
+
+  const parentTagRelIds = childTagRelPostIds.map(_ => uuid({}).slice(0, 17));
+  
+  await updateMutator({
+    collection: Tags,
+    documentId: childTag._id,
+    set: { parentTagId: parentTag._id },
+    validate: false,
+  });
+
+  await TagRels.rawCollection().bulkWrite(childTagRelPostIds.map((postId, i) => ({
+    insertOne: {
+      document: {
+        _id: parentTagRelIds[i],
+        postId: postId,
+        tagId: parentTag._id,
+        createdAt: new Date(),
+        baseScore: 1,
+        afBaseScore: 1,
```

Comment:
I think it would be good to add a boolean field called `backfilled` or something to make it easier to undo these if we want to. It's fairly likely that we will end up thinking a few of the parent/subtags relationships aren't quite right and want to change them. It's fine to not build in the functionality to actually reverse this at this point (although that would be good!) but having a field to keep track of would make it a lot easier to do when needed

Reply:
Great idea, added in https://github.com/ForumMagnum/ForumMagnum/pull/6604/commits/781d41f38f493c6f69e615f7e5e2c969aaaa7711